File: c/LeetCode_108_ConvertSortedArraytoBinarySearchTree.c
#include <stdio.h>
#include <stdlib.h>


struct TreeNode {
    int val; 
    struct TreeNode *left;
    struct TreeNode *right;
};


struct TreeNode* create(int *nums, int begin, int end) {
    if (begin > end) {
        return NULL;
    }

    int inx = (begin + end) / 2;
    
    struct TreeNode *node = (struct TreeNode*) malloc(sizeof(struct TreeNode));
    node->left = NULL;
    node->right = NULL;
    node->val = nums[inx];

    node->left = create(nums, begin, inx - 1);
    node->right = create(nums, inx + 1, end);

    return node;
}


struct TreeNode* sortedArrayToBST(int* nums, int numsSize) {
    if (numsSize == 0 || nums == NULL) {
        return NULL;
    }

    return create(nums, 0, numsSize - 1);
}


int main() {

    return 0;
}

File: python/LeetCode_108_ConvertSortedArraytoBinarySearchTree.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


def sortedArrayToBST(nums):
    def create(nums):
        if not nums:
            return None
        
        lngth = len(nums)
        mid = lngth - (lngth // 2) if lngth % 2 == 0 else lngth // 2
        tree = TreeNode(nums[mid])

        tree.left = create(nums[0:mid])
        tree.right = create(nums[mid+1:lngth])
        return tree

    tree = create(nums)
    return tree

print(sortedArrayToBST([-10,-3,0,5,9,10]))